openedx/adg/lms/applications/tests/test_api/test_views.py

```diff
@@ -349,3 +354,35 @@ def test_is_work_experience_not_applicable_api_with_false(user_application, work
     """
     response = work_experience_not_applicable_api('false')
     assert response.status_code == status.HTTP_200_OK
+
+
+@pytest.mark.django_db
+def test_reference_view_set_create(user_application, login_client):
+    """
+    Verify that the create action of ReferenceViewSet allows adding references only upto the max reference limit against
+    a user application
+    """
```

Comment:
We could also test that the required attributes are provided or not. And if they are, is the validation working as it should.

Reply:
Well, in this case, everything is handled by the viewset provided by Django. We are using ModelViewset. If the required attributes are not provided, a validation error would be raised by the serializer, and I'm not customizing the serializer, just using the ModelSerializer. The only thing I'm customizing in the entire API, is the create action to only keep a check at the max references. So only that needs to be tested.